NitroxClient/GameLogic/Bases/Spawning/BasePiece/BaseWaterParkProcessor.cs

```diff
@@ -30,6 +33,12 @@ protected override void SpawnPostProcess(Base latestBase, Int3 latestCell, GameO
             }
 
             WaterPark waterPark = waterParkPiece.GetWaterParkModule();
+            // When reruning the spawn processor, the module will not be found at first so we need to delay its detection
+            if (!waterPark)
+            {
+                DelayModuleDetection(latestBase, latestCell, finishedPiece);
```

Comment:
Is this a correct call? Should it not be the same as you have before calling a new coroutine? Here you are calling a aync method but not waiting on it. I would have thought a compiler warning would be thrown if nothing else as I believe this is an unadvised call?